### ISSUE TITLE
The getTransaction method throws an error

### ISSUE THREAD

**Issue by Eric-ett** [0xs34n/starknet.js#417]
I use the Starknet4 SDK to send the transaction and call sdk getTransaction, the sdk throws an exception, the error is as follows:
access - Starknet Send After GetTransaction Erro: TypeError: Cannot use 'in' operator to search for 'calldata' in undefined
    at SequencerAPIResponseParser.parseGetTransactionResponse (/app/node_modules/starknet/dist/utils/responseParser/sequencer.js:43:67)
    at /app/node_modules/starknet/dist/provider/sequencer.js:294:53
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (internal/process/task_queues.js:95:5)
    at async StarknetHelp.signTransfer (/app/dist/service/starknet/helper.js:129:30)
    at async Object.sendConsumer [as consumer] (/app/dist/util/maker/send.js:370:30)
    at async promise (/app/dist/util/maker/send_queue.js:26:42)
    at async Promise.all (index 0)
    at async SendQueue.start (/app/dist/util/maker/send_queue.js:37:13)

**Comment by penovicp** [comment]
@Eric-ett, after trying out `getTransaction` I can confirm that it doesn't work as expected for unsuccessful retrievals. This is something that will need to be fixed.

However that fix doesn't need to be a blocker for you, if the transaction hash parameter is valid `getTransaction` behaves as expected.

To corroborate with the base URL set to `https://alpha4.starknet.io`:
- using a faulty hash like `0x000000000000000000000000000000000000000000000000000000000000000` will produce the reported error
- using a valid hash like `0x1c104329cc40c94992c816a6c07e8cedd287d5f7c1d901bb7855923e169a00d` will successfully retrieve the corresponding data

**Comment by Eric-ett** [comment]
Hi @penovicp, I understand that passing an invalid hash will trigger an error like this.
However, my usage scenario is this:
first, I sent a transaction, got the returned txHash, and then I called the getTransaction method, but sometimes it succeeded, and sometimes it returned the above problem. I am sure this txHash is correct, because it will be confirmed on the chain later.
My guess is that after I sent the transaction, the sequencer transaction has not been processed yet, so I cannot get the information of this transaction. But I tried dealy 1 second, 5 seconds, 10 seconds to use getTranscation, this problem still occurs occasionally.
So I'm asking for help, wondering if there is a better solution to this problem
